### PR TITLE
feat(service): Add line service

### PR DESCRIPTION
**Pull request by yoelsusanto**
Changelog:
- Add line service implementation
- Add line service on main readme
- Add line service usage example

Closes: #27

### CODE REVIEW

File: service/line/usage.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  notifier := notify.New()
+  // Tell our notifier to use the line service. You can repeat the above process
+  // for as many services as you like and just tell the notifier to use them.
+  notifier.UseServices(lineService)
+
+  // Send a message
+  err := notifier.Send(context.Background(),
+    "Welcome",
+    "I am a bot written in Go!",
+  )
```
Comment (nikoksr) [inline service/line/usage.md:37]:
```suggestion
  err := notifier.Send(
    context.Background(),
    "Welcome",
    "I am a bot written in Go!",
  )
```

File: service/line/usage.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  // Assuming you already have a line messaging API credential
+  // Provide your channel secret and access token
+  lineService, _ := line.New("channelSecret", "channelAccessToken")
+
+  // Add id from various receivers
+  // You can try to use your own line id for testing
+  lineService.AddReceivers("userID1", "groupID1")
+
+  notifier := notify.New()
+  // Tell our notifier to use the line service. You can repeat the above process
```
Comment (nikoksr) [inline service/line/usage.md:29]:
```suggestion
  notifier := notify.New()
  
  // Tell our notifier to use the line service. You can repeat the above process
```

File: service/line/line.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+func New(channelSecret, channelAccessToken string) (*Line, error) {
+	bot, err := linebot.New(channelSecret, channelAccessToken)
+	if err != nil {
+		return nil, err
+	}
+
+	l := &Line{
+		client: bot,
+	}
+	return l, nil
```
Comment (nikoksr) [inline service/line/line.go:28]:
```suggestion
	}
	
	return l, nil
```

File: service/line/line.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	lineMessage := &linebot.TextMessage{
+		Text: subject + "\n" + message,
+	}
+
+	for _, destinationID := range l.receiverIDs {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+
+		default:
```
Comment (nikoksr) [inline service/line/line.go:48]:
```suggestion
			return ctx.Err()
		default:
```

File: service/line/line.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+			_, err := l.client.PushMessage(destinationID, lineMessage).WithContext(ctx).Do()
+			if err != nil {
+				return errors.Wrapf(err, "failed to send message to line. destination id: '%s'", destinationID)
```
Comment (nikoksr) [inline service/line/line.go:51]:
```suggestion
				return errors.Wrapf(err, "failed to send message to LINE contact '%s'", receiverID)
```

**Review by nikoksr — CHANGES_REQUESTED** [review]
Appreciate your contribution! Just some minor changes and pretty much all related to code style. I would say tests are missing but since most are done yet anyway, I think its better when all missing tests will be done at the same time. So yeah, good job but please fix the minor change requests.

**Comment by nikoksr** [comment]
Hey @yoelsusanto, I just wanted to check in and see if maybe you need help with something or one of my instructions might not be completely understandable. Just let me know if I can help you in any way.

**Comment by yoelsusanto** [comment]
> Hey @yoelsusanto, I just wanted to check in and see if maybe you need help with something or one of my instructions might not be completely understandable. Just let me know if I can help you in any way.

Hi @nikoksr, your instruction has been clear. I just added your requested changes. Let me know if you need anything else.

**Review by nikoksr — APPROVED** [review]
Great, thanks for your contribution and applied changes.

**Pull request merged**